phone-book/src/contacts.js

```diff
@@ -1,39 +1,73 @@
-
-
 class Contacts {
-  constructor(containerSelector) {
-    this.container = document.body.querySelector(containerSelector);
+  constructor(app) {
+    this.title = "Contacts";
+    this.app = app;
+    this.appContainer = app.appContainer;
     this.pageName = 'Contacts';
-    this.users = users;
-    this.usersOrderDirection = 'asc';
-    this.container.addEventListener('click', (event) => {
+    this.users = [];
+    this.usersOrderDirection = 'asc';    
+  }
+
+  initEvents(){
+    this.appContainer.querySelector("main > div.container").addEventListener('click', (event) => {
       if(event.target.nodeName === 'TH'){
         let sortBy = event.target.getAttribute("data-name");
         this.users = this.sortUsersBy(sortBy);
         this.usersOrderDirection = this.usersOrderDirection == 'asc' ? 'desc' : 'asc';
-        this.insertListOfUsers();
+        this.insertListOfUsersToApp(this.users);
       }
-    });
-
-
+    }); 
+    this.getListOfUsers();  
+    this.initSearch(); 
+  }
 
+  showUserPage(id){
+    let promisWithUsers = [];
+    const fetcher = new Fetcher();
+    promisWithUsers.push(fetcher.getUser(id));
+    Promise.all(promisWithUsers).then( (user) => {
+      
+    });
   }
 
+  
   initSearch(){
     document.getElementById('search').addEventListener('keyup', (event) => {
-      this.users = users;
+      let filteredUsers = this.users;
       if(event.target.value !== ''){
-        this.users = this.findUser(event.target.value);
+        filteredUsers = this.findUser(event.target.value);
       }
-      this.insertListOfUsers();
+      this.insertListOfUsersToApp(filteredUsers);
     });    
   }
-  renderListOfUsers(){
-    return this.users.map( user => {
+
+  getListOfUsers(){
+    if(this.users.length != 0 ){
+      return false;
+    }
+    const serverAPI = this.app.serverAPI;
+    const loadingUsers = serverAPI.getUsers();
+    loadingUsers.then( response => {
+      return response.json();
```

Comment:
such transformation is an application layer. What if one day in new spec `response.json()` would be automatical.
Or your Team Lead proposes to you use a axios instead of fetch, or anything else.

In such case, you have to update Contacts page file.

But contacts page know a lot about business logic and should be independent from implementation.

I mean please move transformation to JSON to getUsers method